hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -1935,6 +1937,10 @@ public void shutdown() {
       }
     }
 
+    if (storageLocationChecker != null) {
+      storageLocationChecker.shutdownAndWait(10, TimeUnit.SECONDS);
```

Comment:
Good point. We can exit much quicker since a stalled disk check need not block shutdown at all. I'll change it to 1 second.